engine/api/repositoriesmanager/repositories_manager.go

```diff
@@ -28,23 +28,15 @@ var (
 //CDS private keys in repositories manager have to be stored as secrets in Vault
 //For instance for a repositories manager named "github.com/ovh", the private key
 //is stored in a secret name "repositoriesmanager-secrets-github.com/ovh-privateKey"
-func Initialize(vaultAPI, vaultTokenHeader, vaultKey, vaultPassword, keysDirectory, secretsDir, uiBaseURL, apiBaseURL string) error {
+func Initialize(secretClient secretbackend.Driver, keysDirectory, uiBaseURL, apiBaseURL string) error {
 	uiURL = uiBaseURL
 	apiURL = apiBaseURL
 
 	db := database.DB()
 	if db != nil {
-		vaultClient, err := vault.GetClient(vaultAPI, vaultKey, vaultPassword, keysDirectory, secretsDir, vaultTokenHeader)
-		if err != nil {
-			log.Warning("RepositoriesManager> Unable to get a Vault client %s\n", err)
-			return err
-		}
-
-		secrets, err := vaultClient.GetSecrets()
-
-		if err != nil {
-			log.Warning("RepositoriesManager> Unable to fetch Vault secrets %s\n", err)
-			return sdk.ErrSecretKeyFetchFailed
+		secrets := secretClient.GetSecrets()
```

Comment:
go-plugin seems to be quite unstable with double return values
one of this days i should go through protobuf and grpc...